server/src/test/java/org/elasticsearch/snapshots/mockstore/MockEventuallyConsistentRepositoryTests.java

```diff
@@ -164,13 +167,14 @@ public void testOverwriteSnapshotInfoBlob() throws Exception {
                     Version.CURRENT, Function.identity(), f));
 
             // We try to write another snap- blob for "foo" in the next generation. It fails because the content differs.
-            final AssertionError assertionError = expectThrows(AssertionError.class,
+            assertThatThrownBy(
                 () -> TestFutureUtils.<RepositoryData, Exception>get(f ->
                     repository.finalizeSnapshot(ShardGenerations.EMPTY, 0L, Metadata.EMPTY_METADATA,
                         new SnapshotInfo(snapshotId, Collections.emptyList(),
                             0L, null, 1L, 6, Collections.emptyList(), true),
-                        Version.CURRENT, Function.identity(), f)));
-            assertThat(assertionError.getMessage(), equalTo("\nExpected: <6>\n     but: was <5>"));
+                        Version.CURRENT, Function.identity(), f))
+            ).isInstanceOf(AssertionError.class)
```

Comment:
nit: close `)` on previous line, to keep consistent format for those assertions.